lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+end
+
+--[[
+  NAME
+    dutils.split_filepath - split a filepath into parts
+
+  SYNOPSIS
+    result = dtutils.split_filepath(path_string)
+      filepath - path and filename
+
+  DESCRIPTION
+    split_filepath splits a filepath into the path, filename, basename and filetype and puts
+    that in a table
+
+  RETURN VALUE
+    result - a table containing the path, filename, basename, and filetype
+
+  ERRORS
+
+
+
+]]
+
+function dtutils.split_filepath(str)
+  local result = {}
+  -- Thank you Tobias Jakobs for the awesome regular expression, which I tweaked a little
+  result["path"], result["filename"], result["basename"], result["filetype"] = string.match(str, "(.-)(([^\\/]-)%.?([^%.\\/]*))$")
+  return result
+end
+
+--[[
+  NAME
+    dutils.get_path - get the path from a file path
+
+  SYNOPSIS
+    result = dtutils.get_path(filepath)
+      filepath - path and filename
+
+  DESCRIPTION
+    get_path strips the filename and filetype from a path and returns the path
+
+  RETURN VALUE
+    result - the path
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.get_path(str)
+  local parts = dtutils.split_filepath(str)
+  return parts["path"]
+end
+
+--[[
+  NAME
+    dutils.get_filename - get the filename and extension from a file path
+
+  SYNOPSIS
+    result = dtutils.get_filename(filepath)
+      filepath - path and filename
+
+  DESCRIPTION
+    get_filename strips the path from a filepath and returns the filename
+
+  RETURN VALUE
+    result - the file name and type
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.get_filename(str)
+  local parts = dtutils.split_filepath(str)
+  return parts["filename"]
+end
+
+--[[
+  NAME
+    dutils.get_basename - get the filename without the path or extension
+
+  SYNOPSIS
+    result = dtutils.get_basename(filepath)
+      filepath - path and filename
+
+  DESCRIPTION
+    get_basename returns the name of the file without the path or filetype
+
+  RETURN VALUE
+    result - the basename of the file
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.get_basename(str)
+  local parts = dtutils.split_filepath(str)
+  return parts["basename"]
+end
+
+--[[
+  NAME
+    dutils.get_filetype - get the filetype from a filename
+
+  SYNOPSIS
+    result = dtutils.get_filetype(filepath)
+      filepath - path and filename
+
+  DESCRIPTION
+    get_filetype returns the filetype from the supplied filepath
+
+  RETURN VALUE
+    result - the filetype
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.get_filetype(str)
+  local parts = dtutils.split_filepath(str)
+  return parts["filetype"]
+end
+
+--[[
+  NAME
+    dutils.checkIfBinExists - check if an executable is in the path
+
+  SYNOPSIS
+    result = dtutils.checkIfBinExists(bin)
+      bin - the binary to check for
+
+  DESCRIPTION
+    checkIfBinExists checks to see if the specified binary executable is
+    in the path.
+
+  RETURN VALUE
+    result - true if the executable was found, false if not
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.checkIfBinExists(bin)
```

Comment:
I went through the scripts and pulled what I thought would be useful.  I added a few more as I was writing the plugins.  Unfortunately I was so deep into the trees that I didn't see the forest.  We'll go with the lua convention